### DISCUSSION TITLE
Obstacles to the ship goal for ones not familiar with the stack

### DISCUSSION THREAD

**Discussion by JennieJi** [epicweb-dev/epic-stack#91] [General]
Hi all, 
I am attracted by the goal of EpicStack - ship with less distractions. It is indeed frustrating to select stacks when I want to build MVP to test some ideas.
Despite that the MVP is better to start with something I familiar, I tried with EpicStack which I found most part of the stack were unfamiliar to me (including remix). And the result was:
- day 1: lost in the example files and docs - did nothing but tweeking details to confirm whether things work as I thought.
   - Adopt concept of the full-stack in Remix from the split client, server thinking.
   - Tried to figure out the route rules. I still cannot find the `+` usage in Remix doc.
   - Browsed through Radix docs, that is new to me as well.
- day 2: started to build a simple form but also started to think of give up
  - First thing I found was no hot-reload. `remix dev` should be watching files and rebuild, but instead nothing happened. And the restarting took so so long. This largely blocked me from trying out the stack.
  - Since the form in Radix is marked as `preview`, I took a look at the login example component. Then I found the form is using another library called conform. OK, another new doc to read.
  - 
My little journey was telling me other than choosing stack, using many new things are the biggest enemy of shipping stuff as well. Do you have suggestions?

**Comment by kentcdodds** [comment]
Hi @JennieJi!

Thanks for giving the stack a try! It's still pretty early days for this stack so while I'm disappointed to not give you a stellar experience, I'm not very surprised people unfamiliar with the stack are having a rough time. Especially when you're unfamiliar with the tech, there's a lot to learn up front with any stack. My hope is the docs and tech itself will become much more seamless in the coming weeks and months.

Because you took the time to document your experience here which I greatly appreciate, let me help you with each of your points:

> Adopt concept of the full-stack in Remix from the split client, server thinking.

Yep, there's going to be a learning curve for Remix. Eventually there will be great tutorials on the EpicStack specifically, but until then, the Remix quick start guide is a good resource: https://remix.run/docs/en/main/tutorials/blog (this also includes a free video course by myself if you're interested: https://rmx.as/egghead-course).

> Tried to figure out the route rules. I still cannot find the + usage in Remix doc.

I'll add docs to the Epic Stack. We're using https://github.com/kiliman/remix-flat-routes which allows for the hybrid routes approach.

> Browsed through Radix docs, that is new to me as well.

Yep, more new things to learn. That should only be necessary when you need to modify the existing component code or add new ones. Probably not something you need to review from the start.

> First thing I found was no hot-reload. remix dev should be watching files and rebuild, but instead nothing happened. And the restarting took so so long. This largely blocked me from trying out the stack.

Sadly I think you got a version of the template where I broke this. It should be fixed now.

> Since the form in Radix is marked as preview, I took a look at the login example component. Then I found the form is using another library called conform. OK, another new doc to read.

Yep, I still need to evaluate whether the Radix form component makes sense in this stack or not. Conform offers more than Radix's form component could I think because it includes the server side submission parsing aspect which gives us type safe progressively enhanced forms. It's a joy to work with once you get over the small bump of learning it (and [Zod](https://zod.dev/)).

> My little journey was telling me other than choosing stack, using many new things are the biggest enemy of shipping stuff as well. Do you have suggestions?

Using many things is definitely a trade-off. I could also just build all these things myself so they're all cohesive, but then everyone would have to wait on me and contributors to build what already exists in pieces throughout the ecosystem and also *everyone* would have to learn the Epic Stack version of all of these things rather than being able to rely on the knowledge and experience they have with some of the tools. So I think we're making the right trade-off here.

Thanks for giving the stack a try. I'm going to add some documentation on routing right now :)

**Reply by JennieJi** [reply]
Thanks Kent for the helpful information, more documents surely improve the developer experience!

And appologies about this:
> First thing I found was no hot-reload. remix dev should be watching files and rebuild, but instead nothing happened. And the restarting took so so long. This largely blocked me from trying out the stack.

The day after I wrote down these, I tried another setup and realize some stuff did not work due to my current Windows + Bash on Unbuntu setup 🤦. I probably should try again on clould IDEs instead.

**Reply by JennieJi** [reply]
> Thanks Kent for the helpful information, more documents surely improve the developer experience!
> 
> And appologies about this:
> 
> > First thing I found was no hot-reload. remix dev should be watching files and rebuild, but instead nothing happened. And the restarting took so so long. This largely blocked me from trying out the stack.
> 
> The day after I wrote down these, I tried another setup and realize some stuff did not work due to my current Windows + Bash on Unbuntu setup 🤦. I probably should try again on clould IDEs instead.

I confirmed it was my environment problem. I restarted work with the stack again happily in GitHub CodeWorkspace 😄

**Reply by kentcdodds** [reply]
I'm glad to hear you're happy 🎉
